core/log/file_console_logger.odin

```diff
@@ -31,44 +34,158 @@ Default_File_Logger_Opts :: Options{
 	.Procedure,
 } | Full_Timestamp_Opts
 
+@private
+File_Type :: enum {
+	console,
+	os,
+	os2,
+}
 
+/*
+	Implements a logger that writes to a file stream, such as stdout, or a file on the system.
```

Comment:
Please unindent the doc comments. The indented lines in markdown mark a code block, which is interpreted by Odin's documentation generator as well as other tools (such as editor extensions) as a block of code.

Reply:
Also I would avoid overdocumenting these types. You can look at the allocators in `core:mem` to see how I dealt with documenting allocators there, but basically the `_Data` structrs don't need more documentation than to say which objects they hold data for.

The documentation generator will make the correct links, and if you're using a text editor you'd never use this type so you'd never see its documentation. So the correct course of action is to put the documentation that describes that this is data for console and file loggers, and in the procedures that create them do a more detailed documentation.